### PR TITLE
RecIndex: fix sorting by attributes

### PR DESCRIPTION
**Pull request by cydanil**
Hi,
In this PR, I'm trying to learn my way around the code base. 
I first fixed the display of rating stars.
However, these are not refreshed dynamically yet (the application needs to be relaunched to display the new rating), but this is due to greater issues (the event loop get stuck elsewhere, which must be fixed in a rabbit hole type-of-way).

This also resolves the sorting by attributes in `RecIndex`:

gourmet_sort

I'd welcome some feedback to know whether I'm generally on track or not.
Thanks :)

### CODE REVIEW

**Comment by maweki** [comment]
A few fixes are just whitespace. Try to keep them to a minimum, so we can track where we might have changed logic.

Edit: please don't add a comit reverting that. If you want to fix it (not necessary, just for the future), then you have to rewrite the commits.

**Comment by cydanil** [comment]
Thanks @maweki, I get the point. That's the sort of feedback I am after.

**Comment by kirienko** [comment]
This look a bit strange on my machine:

gourmet-pull-13

Sometimes stars disappear when I hoover over them. But not always. And also sometimes I see some pixel garbage instead of stars.

Though I agree with your changes (I did a half of them trailing another goal).

**Comment by maweki** [comment]
> This look a bit strange on my machine:

This looks like the functionality to actually rate something, right? That widget is probably very broken.

**Comment by kirienko** [comment]
But what I would do now (I mean in general) is I'd merge all these useful fixes here and there. We need to collect a critical mass of positive changes to make the app working more or less stable. It's impossible to fix all these connected changes in one MR.

We just need to be aware of two things: not to merge obvious bullshit, and to explain each other our findings.

Do you agree with that way of development, guys?

**Comment by maweki** [comment]
> But what I would do now (I mean in general) is I'd merge all this useful fixes here and there. We need to collect a critical mass of positive changes to make the app working more or less stable. It's impossible to fix all these connected changes in one MR.
> 
> We just need to be aware of two things: not to merge obvious bullshit, and to explain each other our findings.
> 
> Do you agree with that way of development, guys?

As we do have a "working" codebase, every small fix is fine, even if something else doesn't work quite yet.

File: ui/recCardDisplay.ui [hunk 1]
```diff
@@ -122,7 +122,6 @@
                 <property name="events">GDK_POINTER_MOTION_MASK | GDK_POINTER_MOTION_HINT_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK</property>
                 <property name="show_tabs">False</property>
                 <property name="show_border">False</property>
-                <property name="tab_hborder">1</property>
```
Comment (kirienko) [inline ui/recCardDisplay.ui:125]: In our collection of obsolete things: this `tab_hborder` property is no longer exists.

**Comment by cydanil** [comment]
That seems like a good workflow for the time being, yes.

**Pull request merged**